torch/_dynamo/symbolic_convert.py

```diff
@@ -365,30 +367,10 @@ def wrapper(self: "InstructionTranslatorBase", inst: Instruction):
 
             for _ in range(push):
                 self.push(UnknownVariable())
-
-            resume_call_insts = self.create_call_resume_at(self.next_instruction)
-            # Check if there is a block stack entry with GradModeVariable. And
-            # wrap the instruction causing the graph break inside a try..finally
-            # block. See more details at
-            # https://github.com/pytorch/torchdynamo/issues/207
-            cleanup = []
-            if len(self.block_stack) == 1 and isinstance(
-                self.block_stack[0].with_context, GradModeVariable
-            ):
-                ctx_variable = self.block_stack[0].with_context
-
-                cg = PyCodegen(self)
-                setup_finally, cleanup = ctx_variable.reconstruct(
-                    cg, resume_call_insts[0]
-                )
-                self.output.add_output_instructions(setup_finally)
-
```

Comment:
How about we handle resuming from graph break with a global `ContextManagerManager` (global as in stored on the `OutputGraph`?). We will keep track of live context managers and their `__init__` attributes (i.e. `target_values`) any any given time. 

When executing the code, we install the context setup into each fragment of code that we want to trace. Note that the teardown is already handled by `WithExitFunctionVariable`. 

Furthermore, note that when compiling a function that has graph breaks and `has_tensor_in_frame` (for our purposes, let us call it a `traceable non-inlined code fragment`) the function's execution is already dependent upon which context managers are live at any given moment. 

I'm not super clear how we handle `_set_grad_enabled` for now (seemingly via a large unmaintainable collection of hacks scattered throughout dynamo) so I'll update.

Anyway, I'll try to clarify to what degree the `ContextManager` should be seen as a source of truth. For instance, in the `_is_grad_enabled` cases, it's not clear to me what the behaviour should be when you manually set this variable while a context manager is live, in particular, upon exit of that manager.

Reply:
> How about we handle resuming from graph break with a global ContextManagerManager (global as in stored on the OutputGraph?). We will keep track of live context managers and their __init__ attributes (i.e. target_values) any any given time. (rather than only storing in block_stack)

Another idea for how to do this more dynamically (and without resorting to global dat structures) is to make usage of the stack assuming a function call. Whenever we fail to inline a function given a function call, we trap into a prefix that sets up the stack and creates a a function call with given number of args.

Reply:
Anw, all of what we are doing assumes that context managers are stateless and idempotent - meaning that the source of truth is their `target_values` and nesting them is idempotent.

Reply:
Actually, I've found the perfect solution that does not involve global variables. Everytime we cannot inline a function, instead of defining a global context manager manager, we can just wrap the function call's code in a wrapper that first instantiates our live context managers, install the resultant code object as a global, and then replace that function call with our function call wrapper. 

I think it is a perfect solution and we specialize the function call.

Reply:
The conclusion of the investigations is that most of the solutions detailed were unnecessary and stemmed from a fundamental misunderstanding of the code and its state. We implement a simple solution to the problem which is complete here: https://github.com/pytorch/pytorch/pull/94758.